microservices/first/main.py

```diff
@@ -41,10 +41,18 @@ def start_deployment() -> tuple[Response, int]:
     build and execute. Spawns a new leader in the form of a Docker container
     and execute the SPEED deployment. Returns the id of the newly created
     worker."""
-    url = request.form["url"]
-    branch = request.form["branch"]
+    app.logger.info("start endpoint invoked")
+    data = request.json
+    if data is None:
+        return jsonify({"error": "missing necessary body data"}), 400
```

Comment:
I feel like we should specify that this request requires specifically form body data.

Reply:
I removed the form body and replaced it with JSON. Is form body absolutely required? I'll have to change that back

Reply:
Actually I suppose not. Should be okay as a JSON instead.